internal/component/prometheus/remote/queue/serialization/appender.go

```diff
@@ -0,0 +1,121 @@
+package serialization
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/alloy/internal/component/prometheus/remote/queue/types"
+	"github.com/prometheus/prometheus/model/exemplar"
+	"github.com/prometheus/prometheus/model/histogram"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/model/metadata"
+	"github.com/prometheus/prometheus/storage"
+)
+
+type appender struct {
+	ctx    context.Context
+	ttl    time.Duration
+	s      types.Serializer
+	logger log.Logger
+}
+
+func (a *appender) AppendCTZeroSample(ref storage.SeriesRef, l labels.Labels, t, ct int64) (storage.SeriesRef, error) {
+	// TODO @mattdurham figure out what to do here later. This mirrors what we do elsewhere.
+	return ref, nil
+}
+
+// NewAppender returns an Appender that writes to a given serializer. NOTE the Appender returned writes
+// data immediately and does not honor commit or rollback.
+func NewAppender(ctx context.Context, ttl time.Duration, s types.Serializer, logger log.Logger) storage.Appender {
+	app := &appender{
+		ttl:    ttl,
+		s:      s,
+		logger: logger,
+		ctx:    ctx,
+	}
+	return app
+}
+
+// Append metric
+func (a *appender) Append(ref storage.SeriesRef, l labels.Labels, t int64, v float64) (storage.SeriesRef, error) {
+	// Check to see if the TTL has expired for this record.
+	endTime := time.Now().Unix() - int64(a.ttl.Seconds())
+	if t < endTime {
+		return ref, nil
+	}
+	ts := types.GetTimeSeriesBinary()
+	ts.Labels = l
+	ts.TS = t
+	ts.Value = v
+	ts.Hash = l.Hash()
+	err := a.s.SendSeries(a.ctx, ts)
```

Comment:
Is it guaranteed that `ts` will be returned eventually to the object pool? Would we have a leak if, e.g. the component was removed from Alloy config? I don't see any issues, but would be nice to make this code a bit more clear that this is what's going on, with naming or comments.

Reply:
It should be a required that all time series are returned. Though not in this PR this is checked in a future test via `OutStandingTimeSeriesBinary` atomic int. There are end to end tests that ensure at the end of the test this is zero.